### PR TITLE
[SITIONIX-4/4] Create implementation for creation user

### PR DESCRIPTION
**Pull request by m83city**

### CODE REVIEW

File: api-rest/src/test/java/com/sitionix/athssox/controller/UserControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    private UserApiMapper userApiMapper;
+
+    @Test
+    void givenUserDTO_whenCreate_thenReturnCreatedUser() {
+
+        //given
+        final Long givenUserId = null;
+        final Long expectedUserId = 1L;
+
+        final UserDTO userDTO = this.getUserDTO();
```
Comment (vlad-vinskevitch) [inline api-rest/src/test/java/com/sitionix/athssox/controller/UserControllerTest.java]: why you are using real object here, not a mock?

File: application/src/test/java/com/sitionix/athssox/usecase/CreateUserImplTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    void givenUser_whenCreateUser_thenReturnCreatedUser() {
+
+        //given
+        final Long givenUserId = null;
+        final String givenPassword = "password";
+
+        final Long createdUserId = 1L;
+        final String createdUserPassword = null;
+
+        final User givenUser = this.getUser(givenUserId, givenPassword);
```
Comment (vlad-vinskevitch) [inline application/src/test/java/com/sitionix/athssox/usecase/CreateUserImplTest.java]: why you are using real object here, not a mock?

File: domain/src/main/java/com/sitionix/athssox/domain/User.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.sitionix.athssox.domain;
+
+import lombok.*;
+import org.springframework.stereotype.Component;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Component
```
Comment (vlad-vinskevitch) [inline domain/src/main/java/com/sitionix/athssox/domain/User.java]: why do you need this annotation here @Component

File: infrastructure/postgresql/src/test/java/com/sitionix/athssox/repository/UserRepositoryImplTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    void givenUser_thenCreateUser_thenReturnCreatedUser() {
+
+        //given
+        final Long givenUserId = null;
+        final String givenUserPassword = "password";
+
+        final Long createdUserId = 1L;
+        final String createdUserPassword = "null";
+
+        final User givenUser = this.getUser(givenUserId, givenUserPassword);
```
Comment (vlad-vinskevitch) [inline infrastructure/postgresql/src/test/java/com/sitionix/athssox/repository/UserRepositoryImplTest.java]: why you are using real object here, not a mock?

File: domain/src/main/java/com/sitionix/athssox/domain/User.java (hunk 3 above)
Comment (m83city) [reply domain/src/main/java/com/sitionix/athssox/domain/User.java]: Remove annot

File: infrastructure/postgresql/src/test/java/com/sitionix/athssox/repository/UserRepositoryImplTest.java (hunk 4 above)
Comment (m83city) [reply infrastructure/postgresql/src/test/java/com/sitionix/athssox/repository/UserRepositoryImplTest.java]: resolve

File: api-rest/src/test/java/com/sitionix/athssox/controller/UserControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+        when(this.userApiMapper.asUser(userDTO)).thenReturn(user);
+        when(this.userApiMapper.asUserResponseDTO(user)).thenReturn(expectedBody);
+        when(this.createUser.execute(user)).thenReturn(user);
+
+        //when
+        final ResponseEntity<UserResponseDTO> actualResponse = this.userController.createUser(userDTO);
+
+        //then
+        assertThat(expectedResponse.getStatusCode()).isEqualTo(actualResponse.getStatusCode());
```
Comment (vlad-vinskevitch) [inline api-rest/src/test/java/com/sitionix/athssox/controller/UserControllerTest.java]: why you cannot compare expected and actual?

File: api-rest/src/test/java/com/sitionix/athssox/controller/UserControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Autowired
+    private UserController userController;
+
+    @MockBean
+    private CreateUser createUser;
+
+    @MockBean
+    private UserApiMapper userApiMapper;
+
```
Comment (vlad-vinskevitch) [inline api-rest/src/test/java/com/sitionix/athssox/controller/UserControllerTest.java:46]: add aftter each test verifyNoMoreInteraction()

File: application/src/test/java/com/sitionix/athssox/usecase/CreateUserImplTest.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        }
+    }
+
+    @MockBean
+    private UserRepository userRepository;
+
+    @Autowired
+    private CreateUserImpl createUser;
+
+
```
Comment (vlad-vinskevitch) [inline application/src/test/java/com/sitionix/athssox/usecase/CreateUserImplTest.java]: remove extra spaace

**Review by vlad-vinskevitch — APPROVED** [review]

**Pull request merged**